test/example/stats_distribution_normal/demo_normal_rvs.f90

```diff
@@ -0,0 +1,45 @@
+program demo_normal_rvs
+    use stdlib_random, only: random_seed
+    use stdlib_stats_distribution_normal, only: norm => rvs_normal
+
+    implicit none
+    real ::  a(2, 3, 4), b(2, 3, 4)
+    complex :: loc, scale
+    integer :: seed_put, seed_get
+
+    seed_put = 1234567
+    call random_seed(seed_put, seed_get)
+
+    print *, norm()           !single standard normal random variate
+
+! 0.563655198
+
+    print *, norm(1.0, 2.0)
+!normal random variate mu=1.0, sigma=2.0
+
+! -0.633261681
+
+    print *, norm(0.0, 1.0, 10)       !an array of 10 standard norml random variates
+
+! -3.38123664E-02  -0.190365672  0.220678389  -0.424612164  -0.249541596
+!  0.865260184  1.11086845  -0.328349441  1.10873628  1.27049923
+
+    a(:, :, :) = 1.0
+    b(:, :, :) = 1.0
+    print *, norm(a, b)         ! a rank 3 random variates array
```

Comment:
```
$ gh pr checkout 662
$ ./ci/fpm-deployment.sh && cd stdlib-fpm
$ fpm test
demo_normal_rvs.f90                    failed.
[   0%]Compiling...
example/demo_normal_rvs.f90:29:12:

   29 |     print *, norm(a, b)         ! a rank 3 random variates array
      |            1
Error: There is no specific function for the generic ‘rvs_normal’ at (1)
<ERROR> Compilation failed for object " example_demo_normal_rvs.f90.o "
<ERROR>stopping due to failed compilation
STOP 1
```

I got this error while trying the PR with fpm, and it seems the `rvs_normal` is missing some `elemental` statements.
The obvious patch below fixes it, should I open another PR just for this one?

```diff
diff --git a/src/stdlib_stats_distribution_normal.fypp b/src/stdlib_stats_distribution_normal.fypp
index e21b7af..5a02805 100644
--- a/src/stdlib_stats_distribution_normal.fypp
+++ b/src/stdlib_stats_distribution_normal.fypp
@@ -160,6 +160,7 @@ contains
 
 
     #:for k1, t1 in REAL_KINDS_TYPES
+    impure elemental &
     function rvs_norm_${t1[0]}$${k1}$(loc, scale) result(res)
     !
     ! Normal random variate (loc, scale)
@@ -178,6 +179,7 @@ contains
 
 
     #:for k1, t1 in CMPLX_KINDS_TYPES
+    impure elemental &
     function rvs_norm_${t1[0]}$${k1}$(loc, scale) result(res)
     !
     ! Normally distributed complex. The real part and imaginary part are       &
@@ -197,6 +199,7 @@ contains
 
 
     #:for k1, t1 in REAL_KINDS_TYPES
+    impure elemental &
     function rvs_norm_array_${t1[0]}$${k1}$(loc, scale, array_size) result(res)
         ${t1}$, intent(in) :: loc, scale
         integer, intent(in) :: array_size
@@ -250,6 +253,7 @@ contains
 
 
     #:for k1, t1 in CMPLX_KINDS_TYPES
+    impure elemental &
     function rvs_norm_array_${t1[0]}$${k1}$(loc, scale, array_size) result(res)
         ${t1}$, intent(in) :: loc, scale
         integer, intent(in) :: array_size
```

Reply:
> The obvious patch below fixes it, should I open another PR just for this one?

Good catch. Yes, sure!

Reply:
I'm investigating further. My patch didn't work it just skipped this specific error and created another one :)
the generated ones with *array* in the interface shouldn't have elemental.